contracts/supplementary/rebalancing-manager/RebalancingTokenManager.sol

```diff
@@ -0,0 +1,456 @@
+import { IRebalancingSetToken } from "../../core/interfaces/IRebalancingSetToken.sol";
+import { ISetToken } from "../../core/interfaces/ISetToken.sol";
+import { RebalancingHelperLibrary } from "../../core/lib/RebalancingHelperLibrary.sol";
+
+
+/**
+ * @title LinearAuctionPriceCurve
+ * @author Set Protocol
+ *
+ * Contract used to manage a Rebalancing Set Token
+ */
+contract RebalancingTokenManager {
+
+    using SafeMath for uint256;
+    using AddressArrayUtils for address[];
+
+    /* ============ Constants ============ */
+
+    uint256 constant PRICE_PRECISION = 100;
+    uint256 constant AUCTION_LIB_PRICE_DIVISOR = 1000;
+    uint256 constant BTC_DECIMALS = 8;
+    uint256 constant ETH_DECIMALS = 18;
+    uint256 constant SET_TOKEN_DECIMALS = 18;
+    uint256 constant DECIMAL_DIFF_MULTIPLIER = 10**10;
+    uint256 constant THIRTY_MINUTES_IN_SECONDS = 1800;
+
+    /* ============ State Variabales ============ */
+
+    address public btcAddress;
+    address public ethAddress;
+    address public setTokenFactory;
+    IMedian public btcPriceFeed;
+    IMedian public ethPriceFeed;
+    ICore coreInterface;
+
+    address public auctionLibrary;
+    uint256 public auctionTimeToPivot;
+
+    /* ============ Constructor ============ */
+
+    /*
+     * Rebalancing Token Manager constructor
+     *
+     * @param  _coreAddress             The address of the Core contract
+     * @param  _btcPriceFeedAddress     The address of BTC medianize
+     * @param  _ethPriceFeedAddress     The address of ETH medianize
+     * @param  _btcAddress              The address of the wrapped BTC contract
+     * @param  _ethAddress              The address of the wrapped ETH contract
+     * @param  _setTokenFactory         The address of the SetTokenFactory
+     * @param  _auctionLibrary          The address of auction price curve to use in rebalance
+     * @param  _auctionTimeToPivot      The amount of time until pivot reached in rebalance
+     */
+    constructor(
+        address _coreAddress,
+        address _btcPriceFeedAddress,
+        address _ethPriceFeedAddress,
+        address _btcAddress,
+        address _ethAddress,
+        address _setTokenFactory,
+        address _auctionLibrary,
+        uint256 _auctionTimeToPivot
+    )
+        public
+    {
+        coreInterface = ICore(_coreAddress);
+        btcPriceFeed = IMedian(_btcPriceFeedAddress);
+        ethPriceFeed = IMedian(_ethPriceFeedAddress);
+        btcAddress = _btcAddress;
+        ethAddress = _ethAddress;
+        setTokenFactory = _setTokenFactory;
+
+        auctionLibrary = _auctionLibrary;
+        auctionTimeToPivot = _auctionTimeToPivot;
+    }
+
+    /* ============ External ============ */
+
+    /*
+     * When allowed on RebalancingSetToken, anyone can call for a new rebalance proposal
+     *
+     * @param  _rebalancingSetTokenAddress     The address of Rebalancing Set Token to propose new allocation
+     */
+    function proposeNewRebalance(
```

Comment:
`propose`? makes sense to me that this contract should conform to the existing interface